File: backspace_compare_leetcode#844.rb
def backspace_compare(s, t)
    ans1 = ''
    ans2 = ''
    s.each_char do |char|
        if char == '#'
            ans1 = ans1[0...ans1.length-1]
        else
           ans1 += char 
        end
    end
    
    t.each_char do |char|
        if char == '#'
            ans2 = ans2[0...ans2.length-1]
        else
           ans2 += char 
        end
    end
    
    ans1 == ans2
end

File: backspaceCompare_leetcode#844.js
var backspaceCompare = function(S, T) {
    let ans1 = '';
    let ans2 = '';
    let i = 0;
    let j = 0;
    while(i < S.length){
        if(S[i] == '#'){
            ans1 = ans1.slice(0,ans1.length-1)
        } else {
            ans1 += S[i];
        }
        i++
    }
    
    while(j < T.length){
        if(T[j] == '#'){
            ans2 = ans2.slice(0,ans2.length-1)
        } else {
            ans2 += T[j];
        }
        j++
    }
    return ans1 == ans2;
  
};

console.log(backspaceCompare( "xywrrmp", "xywrrmu#p"))